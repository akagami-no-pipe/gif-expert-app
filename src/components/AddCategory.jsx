import { useState } from 'react'

export const AddCategory = () => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
    };

  return (
    <form onSubmit={ onSubmit }>
        <input 
            type="text" 
            placeholder="Buscar gif" 
            value={inputValue} 
            onChange={ onInputChange }
        />
    </form>
  )
}
