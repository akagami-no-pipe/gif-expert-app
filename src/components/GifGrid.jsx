
export const GifGrid = ({category}) => {

    const gifs = [1,2,3,4];

    return (
        <>
            <h3 >{ category }</h3>
            {
                gifs.map( (gif) => (                    
                    <div >{gif}</div>
                ))
            }
        </>
    )

}
