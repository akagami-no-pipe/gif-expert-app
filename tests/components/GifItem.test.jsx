import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas del componente GifItem', () => {

    const url = `https://github.com/`;
    const title = 'Este es un titulo';

    test('Debe validar que url y texto alternativo no venga vacio y sea el indicado', () => {
        render(<GifItem url={url} title={title}/>);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe validar que titulo no venga vacio y se muestre el indicado en el <p></p>', () => {
        // const {getAllByAltText} = render(<GifItem url={url} title={title}/>);
        // expect(getAllByAltText('Este es un titulo')).toBeTruthy();
        render(<GifItem url={url} title={title}/>);
        const {textContent} = screen.getByRole('paragraph');
        expect(textContent).toBe(title);
    });

    test('Probando el snapshot', () => {
        const { container } = render(<GifItem url={url} title={title}/>);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar la imaagen con ell url y el alt indicado', () => { 
        render(<GifItem url={url} title={title}/>);
        // screen.debug();

    });
});