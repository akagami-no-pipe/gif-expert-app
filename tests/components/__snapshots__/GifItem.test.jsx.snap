// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifItem Probando el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Este es un titulo"
      src="https://github.com/"
    />
    <p>
      Este es un titulo
    </p>
  </div>
</div>
`;
